OPAL/tac/src/main/scala/org/opalj/tac/fpcf/analyses/fieldassignability/L2FieldAssignabilityAnalysis.scala

```diff
@@ -245,21 +247,21 @@ class L2FieldAssignabilityAnalysis private[analyses] (val project: SomeProject)
         writes:                     Seq[(DefinedMethod, TACode[TACMethodParameter, V], Option[V], Int)]
     )(implicit state: State): Boolean = {
         writes.exists { writeAccess =>
-            fieldReadAccessInformation.accesses.count { readAccess =>
-                contextProvider.contextFromId(readAccess._1).method eq writeAccess._1
-            } > 1 &&
-                fieldReadAccessInformation.getNewestAccesses(
-                    fieldReadAccessInformation.numDirectAccesses - seenDirectAccesses,
-                    fieldReadAccessInformation.numIndirectAccesses - seenIndirectAccesses
-                ).exists { readAccess =>
-                        val method = contextProvider.contextFromId(readAccess._1).method
-                        (writeAccess._1 eq method) && {
-                            val taCode = state.tacDependees(method.asDefinedMethod).ub.tac.get
-                            val writeIndex = writeAccess._4
-                            val readIndex = taCode.pcToIndex(readAccess._2)
-                            writeIndex != readIndex && !dominates(writeIndex, readIndex, taCode)
+            fieldReadAccessInformation.getNewestAccesses(
+                fieldReadAccessInformation.numDirectAccesses - seenDirectAccesses,
+                fieldReadAccessInformation.numIndirectAccesses - seenIndirectAccesses
+            ).exists { readAccess =>
+                    val method = contextProvider.contextFromId(readAccess._1).method
+                    (writeAccess._1 eq method) && {
+                        val taCode = state.tacDependees(method.asDefinedMethod).ub.tac.get
+
+                        if (readAccess._3.isDefined && readAccess._3.get._2.exists(isFormalParameter)) {
```

Comment:
Shouldn't this be `forall` to ensure soundness?

Reply:
True, will fix.

What do you think of the remaining solution? The tests are still failing, though now it fixes some other cases that were previously wrong imo. I will analyze them bit by bit tonight, but we prolly will just refresh the test targets.